x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/proto/SqlVersions.java

```diff
@@ -0,0 +1,212 @@
+    public static final SqlVersion V_8_3_3 = fromId(8_03_03_99);
+    public static final SqlVersion V_8_4_0 = fromId(8_04_00_99);
+    public static final SqlVersion V_8_4_1 = fromId(8_04_01_99);
+    public static final SqlVersion V_8_4_2 = fromId(8_04_02_99);
+    public static final SqlVersion V_8_4_3 = fromId(8_04_03_99);
+    public static final SqlVersion V_8_5_0 = fromId(8_05_00_99);
+    public static final SqlVersion V_8_5_1 = fromId(8_05_01_99);
+    public static final SqlVersion V_8_5_2 = fromId(8_05_02_99);
+    public static final SqlVersion V_8_5_3 = fromId(8_05_03_99);
+    public static final SqlVersion V_8_6_0 = fromId(8_06_00_99);
+    public static final SqlVersion V_8_6_1 = fromId(8_06_01_99);
+    public static final SqlVersion V_8_6_2 = fromId(8_06_02_99);
+    public static final SqlVersion V_8_7_0 = fromId(8_07_00_99);
+    public static final SqlVersion V_8_7_1 = fromId(8_07_01_99);
+    public static final SqlVersion V_8_8_0 = fromId(8_08_00_99);
+    public static final SqlVersion V_8_8_1 = fromId(8_08_01_99);
+    public static final SqlVersion V_8_8_2 = fromId(8_08_02_99);
+    public static final SqlVersion V_8_9_0 = fromId(8_09_00_99);
+    public static final SqlVersion V_8_9_1 = fromId(8_09_01_99);
+    public static final SqlVersion V_8_9_2 = fromId(8_09_02_99);
+    public static final SqlVersion V_8_10_0 = fromId(8_10_00_99);
+    public static final SqlVersion V_8_10_1 = fromId(8_10_01_99);
+    public static final SqlVersion V_8_10_2 = fromId(8_10_02_99);
+    public static final SqlVersion V_8_10_3 = fromId(8_10_03_99);
+    public static final SqlVersion V_8_10_4 = fromId(8_10_04_99);
+    public static final SqlVersion V_8_11_0 = fromId(8_11_00_99);
+    public static final SqlVersion V_8_11_1 = fromId(8_11_01_99);
+    public static final SqlVersion V_8_11_2 = fromId(8_11_02_99);
+    public static final SqlVersion V_8_11_3 = fromId(8_11_03_99);
+    public static final SqlVersion V_8_11_4 = fromId(8_11_04_99);
+    public static final SqlVersion V_8_12_0 = fromId(8_12_00_99);
+    public static final SqlVersion V_8_12_1 = fromId(8_12_01_99);
+    public static final SqlVersion V_8_12_2 = fromId(8_12_02_99);
+    public static final SqlVersion V_8_13_0 = fromId(8_13_00_99);
+    public static final SqlVersion V_8_13_1 = fromId(8_13_01_99);
+    public static final SqlVersion V_8_13_2 = fromId(8_13_02_99);
+    public static final SqlVersion V_8_13_3 = fromId(8_13_03_99);
+    public static final SqlVersion V_8_13_4 = fromId(8_13_04_99);
+    public static final SqlVersion V_8_14_0 = fromId(8_14_00_99);
+    public static final SqlVersion V_8_14_1 = fromId(8_14_01_99);
+    public static final SqlVersion V_8_14_2 = fromId(8_14_02_99);
+    public static final SqlVersion V_8_14_3 = fromId(8_14_03_99);
+    public static final SqlVersion V_8_15_0 = fromId(8_15_00_99);
+    public static final SqlVersion V_8_15_1 = fromId(8_15_01_99);
+    public static final SqlVersion V_8_16_0 = fromId(8_16_00_99);
```

Comment:
main is now v9, which can only talk to 8.17 nodes (when released). Does this mean we can remove all these versions, or are these actually _client_ versions which still need to be compatible? Is this documented somewhere?

Reply:
> Is this documented somewhere?

The  the current version compatibility policy has been documented [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-jdbc.html#jdbc-compatibility). This has been applying to the 7.x and 8.x.

Given the SQL maintenance status, this PR proposes updating this policy by allowing unrestricted connectivity between clients and servers past 8.16 (comment [here](https://github.com/elastic/elasticsearch/pull/112094/files#diff-35b51352875c23b3a8029fab7ad54b93bb9363c8d5a14bfd61992a534a3e86edR28-R32)). Any pre-8.16 releases (including all 8.15 releases to follow) will have to comply with the exiting policy (linked above).
The SL releases will be considered at version 8.16 (the "server compatibility version").

> Does this mean we can remove all these versions.

The versions should remain in the 8.x branch, but those 7.x not referenced explicitly in `VersionCompatibility` class (I think it's just `SqlVersions.V_7_12_0`) could be removed on 9.x. I think this would be done after back porting to 8.x though.

> are these actually client versions which still need to be compatible?

Yes, the post 7.7.0 client releases would need to stay compatible with 8.x servers.

Reply:
> those 7.x not referenced explicitly in VersionCompatibility class (I think it's just SqlVersions.V_7_12_0) could be removed on 9.x

This could be an `@UpdateForV9` tag on the 7.0.0 version/SqlVersion class.

I think it's worth a descriptive comment at this point, that no further versions should be added in the normal course of things.